homeassistant/components/imap/coordinator.py

```diff
@@ -34,8 +43,17 @@
 
 async def connect_to_server(data: Mapping[str, Any]) -> IMAP4_SSL:
     """Connect to imap server and return client."""
-    client = IMAP4_SSL(data[CONF_SERVER], data[CONF_PORT])
+    if (
+        ciphers := data.get(CONF_SSL_CIPHER_LIST, SSLCipherList.PYTHON_DEFAULT)
+    ) != SSLCipherList.PYTHON_DEFAULT:
+        context = ssl.create_default_context()
+        context.set_ciphers(SSL_CIPHER_LISTS[SSLCipherList(ciphers)])
+        client = IMAP4_SSL(data[CONF_SERVER], data[CONF_PORT], ssl_context=context)
+    else:
+        client = IMAP4_SSL(data[CONF_SERVER], data[CONF_PORT])
+
```

Comment:
you can use the [`client_context`](https://github.com/home-assistant/core/blob/06520217bc2c924e9245be3f0df1197f6977d954/homeassistant/util/ssl.py#L91) from `homeassistant.util.ssl` to get a ssl context - like:

```python
context = client_context(data.get(CONF_SSL_CIPHER_LIST, SSLCipherList.PYTHON_DEFAULT))
```